test/WireMock.Net.Tests/WireMockServerTests.cs

```diff
@@ -115,6 +115,43 @@ public async Task WireMockServer_Should_delay_responses_for_a_given_route()
             server.Stop();
         }
 
+        [Fact]
+        public async Task WireMockServer_Should_randomly_delay_responses_for_a_given_route()
+        {
+            // given
+            var server = WireMockServer.Start();
+
+            server
+                .Given(Request.Create()
+                    .WithPath("/*"))
+                .RespondWith(Response.Create()
+                    .WithBody(@"{ msg: ""Hello world!""}")
+                    .WithRandomDelay(10, 1000));
+
+            // when
+            var watch = new Stopwatch();
+            watch.Start();
+
+            var httClient = new HttpClient();
+            async Task<long> ExecuteTimedRequest()
+            {
+                watch.Reset();
+                await httClient.GetStringAsync("http://localhost:" + server.Ports[0] + "/foo");
+                return watch.ElapsedMilliseconds;
+            }
+
+            var elapsed1 = ExecuteTimedRequest();
+            var elapsed2 = ExecuteTimedRequest();
+            var elapsed3 = ExecuteTimedRequest();
+
+            // then
+            Check.That(elapsed1).IsNotEqualTo(elapsed2)
```

Comment:
In theory it could be that all three values are the same ;-) Which makes you test break.